receiver/googlecloudspannerreceiver/internal/metadata/metricsmetadata.go

```diff
@@ -0,0 +1,223 @@
+	// In addition to common metric labels
+	QueryLabelValuesMetadata  []LabelValueMetadata
+	QueryMetricValuesMetadata []MetricValueMetadata
+}
+
+func (metadata *MetricsMetadata) timestamp(row *spanner.Row) (time.Time, error) {
+	if metadata.MetadataType() == MetricsMetadataTypeCurrentStats {
+		return time.Now().UTC(), nil
+	}
+	var timestamp time.Time
+	err := row.ColumnByName(metadata.TimestampColumnName, &timestamp)
+	return timestamp, err
+}
+
+func (metadata *MetricsMetadata) toLabelValues(row *spanner.Row) ([]LabelValue, error) {
+	values := make([]LabelValue, len(metadata.QueryLabelValuesMetadata))
+
+	for i, metadataItems := range metadata.QueryLabelValuesMetadata {
+		var err error
+
+		if values[i], err = toLabelValue(metadataItems, row); err != nil {
+			return nil, err
+		}
+	}
+
+	return values, nil
+}
+
+func toLabelValue(labelValueMetadata LabelValueMetadata, row *spanner.Row) (LabelValue, error) {
+	valueHolder := labelValueMetadata.ValueHolder()
+
+	err := row.ColumnByName(labelValueMetadata.ColumnName(), valueHolder)
+	if err != nil {
+		return nil, err
+	}
+
+	var value LabelValue
+
+	switch labelValueMetadataCasted := labelValueMetadata.(type) {
+	case StringLabelValueMetadata:
+		value = newStringLabelValue(labelValueMetadataCasted, valueHolder)
+	case Int64LabelValueMetadata:
+		value = newInt64LabelValue(labelValueMetadataCasted, valueHolder)
+	case BoolLabelValueMetadata:
+		value = newBoolLabelValue(labelValueMetadataCasted, valueHolder)
+	case StringSliceLabelValueMetadata:
+		value = newStringSliceLabelValue(labelValueMetadataCasted, valueHolder)
+	case ByteSliceLabelValueMetadata:
+		value = newByteSliceLabelValue(labelValueMetadataCasted, valueHolder)
+	}
+
+	return value, nil
+}
+
+func (metadata *MetricsMetadata) toMetricValues(row *spanner.Row) ([]MetricValue, error) {
+	values := make([]MetricValue, len(metadata.QueryMetricValuesMetadata))
+
+	for i, metadataItems := range metadata.QueryMetricValuesMetadata {
+		var err error
+
+		if values[i], err = toMetricValue(metadataItems, row); err != nil {
+			return nil, err
+		}
+	}
+
+	return values, nil
+}
+
+func toMetricValue(metricValueMetadata MetricValueMetadata, row *spanner.Row) (MetricValue, error) {
+	valueHolder := metricValueMetadata.ValueHolder()
+
+	err := row.ColumnByName(metricValueMetadata.ColumnName(), valueHolder)
+	if err != nil {
+		return nil, err
+	}
+
+	var value MetricValue
+
+	switch metricValueMetadataCasted := metricValueMetadata.(type) {
+	case Int64MetricValueMetadata:
+		value = newInt64MetricValue(metricValueMetadataCasted, valueHolder)
+	case Float64MetricValueMetadata:
+		value = newFloat64MetricValue(metricValueMetadataCasted, valueHolder)
+	}
+
+	return value, nil
+}
+
+func (metadata *MetricsMetadata) RowToMetrics(databaseID *datasource.DatabaseID, row *spanner.Row) ([]pdata.Metrics, error) {
+	timestamp, err := metadata.timestamp(row)
+	if err != nil {
+		return nil, fmt.Errorf("error occurred during extracting timestamp %v", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("error occurred during extracting timestamp: %w", err)
```

Reply:
Same applies to other similar constructs.

Reply:
Thank you for notice. Corrected all such statements.